genie_registry/clinical.py

```diff
@@ -175,6 +175,33 @@ def _check_int_year_consistency(
     return ""
 
 
+def _check_year_death_validity(clinicaldf: pd.DataFrame) -> str:
+    """
+    Check if YEAR_DEATH >= YEAR_CONTACT
+
+    Args:
+        clinicaldf: Clinical Data Frame
+
+    Returns:
+        Error message if YEAR_DEATH if invalid
+    """
+    error = ""
+    # Generate temp dataframe to handle datatype mismatch in a column
+    temp = clinicaldf.copy()
+    # Convert YEAR_DEATH and YEAR_CONTACT to numeric, coercing errors to NaN
+    temp["YEAR_DEATH"] = pd.to_numeric(temp["YEAR_DEATH"], errors="coerce")
+    temp["YEAR_CONTACT"] = pd.to_numeric(temp["YEAR_CONTACT"], errors="coerce")
+    check_result = np.where(
+        (pd.isna(temp["YEAR_DEATH"]) | pd.isna(temp["YEAR_CONTACT"])),
+        "N/A",
+        temp["YEAR_DEATH"] >= temp["YEAR_CONTACT"],
+    )
+    index = [",".join(str(idx)) for idx, i in enumerate(check_result) if i == "False"]
+    if "False" in check_result:
+        error = "Patient Clinical File: Please double check your YEAR_DEATH and YEAR_CONTACT columns. YEAR_DEATH must be >= YEAR_CONTACT.\n"
```

Comment:
Should you include the value(s) in the error message that caused the result to occur?

Reply:
From what I learned, we will eventually move forward to row-wise validation but this is the timeline is TBD.

Reply:
For more context: 

Reasoning why we wouldn't add the row-based validation (i think this is more helpful than just putting in values) in the error message yet just because we haven't decided to start implementing and delivering row-based validation for the centers/sites as a whole and we would have to talk about the standard for doing that so we are consistent across all validation rules. (And most likely discuss with the sites as a whole as well) so bigger discussions to be had there.